news/5333-remove-24.7.x-deprecations

```diff
@@ -0,0 +1,95 @@
+  * `conda_build.conda_interface.normalized_version` (use `conda.models.version.normalized_version` instead)
+  * `conda_build.conda_interface.os` (use `os` instead)
+  * `conda_build.conda_interface.PackageRecord` (use `conda.models.records.PackageRecord` instead)
+  * `conda_build.conda_interface.PaddingError` (use `conda.exceptions.PaddingError` instead)
+  * `conda_build.conda_interface.partial` (use `functools.partial` instead)
+  * `conda_build.conda_interface.PathType` (use `conda.models.enums.PathType` instead)
+  * `conda_build.conda_interface.pkgs_dirs` (use `conda.base.context.context.pkgs_dirs` instead)
+  * `conda_build.conda_interface.prefix_placeholder` (use `conda.base.constants.PREFIX_PLACEHOLDER` instead)
+  * `conda_build.conda_interface.ProgressiveFetchExtract` (use `conda.core.package_cache_data.ProgressiveFetchExtract` instead)
+  * `conda_build.conda_interface.reset_context` (use `conda.base.context.reset_context` instead)
+  * `conda_build.conda_interface.Resolve` (use `conda.resolve.Resolve` instead)
+  * `conda_build.conda_interface.rm_rf` (use `conda_build.utils.rm_rf` instead)
+  * `conda_build.conda_interface.root_dir` (use `conda.base.context.context.root_prefix` instead)
+  * `conda_build.conda_interface.root_writable` (use `conda.base.context.context.root_writable` instead)
+  * `conda_build.conda_interface.spec_from_line` (use `conda.cli.common.spec_from_line` instead)
+  * `conda_build.conda_interface.specs_from_args` (use `conda.cli.common.specs_from_args` instead)
+  * `conda_build.conda_interface.specs_from_url` (use `conda.cli.common.specs_from_url` instead)
+  * `conda_build.conda_interface.StringIO` (use `io.StringIO` instead)
+  * `conda_build.conda_interface.subdir` (use `conda.base.context.context.subdir` instead)
+  * `conda_build.conda_interface.symlink_conda`
+  * `conda_build.conda_interface.TemporaryDirectory` (use `conda.gateways.disk.create.TemporaryDirectory` instead)
+  * `conda_build.conda_interface.TmpDownload` (use `conda.gateways.connection.download.TmpDownload` instead)
+  * `conda_build.conda_interface._toposort` (use `conda.common.toposort._toposort` instead)
+  * `conda_build.conda_interface.unix_path_to_win` (use `conda.utils.unix_path_to_win` instead)
+  * `conda_build.conda_interface.untracked` (use `conda.misc.untracked` instead)
+  * `conda_build.conda_interface.Unsatisfiable` (use `conda.exceptions.UnsatisfiableError` instead)
+  * `conda_build.conda_interface.UnsatisfiableError` (use `conda.exceptions.UnsatisfiableError` instead)
+  * `conda_build.conda_interface.url_path` (use `conda.utils.url_path` instead)
+  * `conda_build.conda_interface.VersionOrder` (use `conda.models.version.VersionOrder` instead)
+  * `conda_build.conda_interface.walk_prefix` (use `conda.misc.walk_prefix` instead)
+  * `conda_build.conda_interface.win_path_to_unix` (use `conda.common.path.win_path_to_unix` instead)
+  * `conda_build.index.channel_data`
```

Comment:
```suggestion
  * `conda_build.index.channel_data`
  * `conda_build.index.get_build_index` return value for `channel_data` is now always `None`
```
or something along those lines maybe?

Reply:
Oh good call, I'll update the news file with this info!